packages/x-scheduler/src/material/internals/components/day-time-grid/TimeGridColumn.tsx

```diff
@@ -33,6 +36,95 @@ export function TimeGridColumn(props: TimeGridColumnProps) {
       data-weekend={isWeekend(adapter, day.value) ? '' : undefined}
       data-current={isToday ? '' : undefined}
       style={{ '--columns-count': maxIndex } as React.CSSProperties}
+    >
+      <ColumnInteractiveLayer
+        day={day}
+        start={start}
+        end={end}
+        showCurrentTimeIndicator={showCurrentTimeIndicator}
+        index={index}
+        occurrences={occurrences}
+        maxIndex={maxIndex}
+      />
+    </TimeGrid.Column>
+  );
+}
+
+function ColumnInteractiveLayer({
+  day,
+  start,
+  end,
+  showCurrentTimeIndicator,
+  index,
+  occurrences,
+  maxIndex,
+}: {
+  day: useEventOccurrencesWithDayGridPosition.DayData;
+  start: SchedulerValidDate;
+  end: SchedulerValidDate;
+  showCurrentTimeIndicator: boolean;
+  index: number;
+  occurrences: CalendarEventOccurrenceWithTimePosition[];
+  maxIndex: number;
+}) {
+  const adapter = useAdapter();
+
+  const { getCursorPositionInElementMs } = useTimeGridColumnContext();
+  const placeholder = TimeGrid.usePlaceholderInRange({ start, end, occurrences, maxIndex });
+  const store = useEventCalendarStoreContext();
+  const columnRef = React.useRef<HTMLDivElement | null>(null);
+  const { startEditing } = useEventPopover();
+  const rawPlaceholder = useStore(store, selectors.occurrencePlaceholder);
```

Comment:
Same, you could return `isCreation` directly in the selector to avoid re-renders